### PR TITLE
MSPSDS-197: Use Keycloak to authenticate users and manage their roles

### PR DESCRIPTION
**Pull request by tom-bell-softwire**
This adds a Keycloak instance to the Docker app configuration and uses the [Keycloak gem](https://github.com/imagov/keycloak) to authenticate users against the Keycloak IdP, fetch their details, determine roles, etc.

It also removes Devise and Rolify and their associated `users` and `roles` tables.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,6 +7,12 @@
     resources :images
   end
 
+  get "sessions/new"
```
[the remark this answers is not in the document]
Comment (tom-bell-softwire) [reply config/routes.rb]: Good call!

File: app/views/sessions/forgot_password.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<% content_for :page_title, "Forgot password - Market Surveillance & Product Safety" %>
```
[the remark this answers is not in the document]
Comment (tom-bell-softwire) [reply app/views/sessions/forgot_password.erb]: Ah, forgot to remove this. Plan is just to use Keycloak's own login page instead.

File: app/controllers/invitations_controller.rb [hunk 3]
```diff
@@ -1,14 +1,12 @@
-class InvitationsController < Devise::InvitationsController
+class InvitationsController
```
[the remark this answers is not in the document]
Comment (tom-bell-softwire) [reply app/controllers/invitations_controller.rb]: Nope! There are also a bunch of Devise-generated views under `views/users/` that should be removed as well.

File: .env-template [hunk 4]
```diff
@@ -1,4 +1,9 @@
 # Copy this file to .env and fill in the values with your own env variables
 # Only needed if you want to test email sending
 NOTIFY_API_KEY=my_key-xxxx
-COMPANIES_HOUSE_API_KEY=xxxx
\ No newline at end of file
+COMPANIES_HOUSE_API_KEY=xxxx
+
+# Keycloak client credentials
+KEYCLOAK_AUTH_URL=
```
[the remark this answers is not in the document]
Comment (broder) [reply .env-template]: Sounds good!

**Pull request merged**